tests/cpp/ir/type_test.cpp

```diff
@@ -18,10 +18,10 @@ TEST(Type, BitTypes) {
   auto qu11 = TypeFactory::get_instance().get_quant_int_type(11, false, i32);
   auto u16 = TypeFactory::get_instance().get_primitive_int_type(16, false);
 
-  auto bs =
-      TypeFactory::get_instance().get_bit_struct_type(u16, {qi5, qu11}, {0, 5});
+  auto bs = TypeFactory::get_instance().get_bit_struct_type(
+      u16, {qi5, qu11}, {0, 5}, {false, false}, {-1, -1}, {{}, {}});
 
-  EXPECT_EQ(bs->to_string(), "bs(qi5@0, qu11@5)");
```

Comment:
nit: how about adding an additional check for bit_struct_type with shared exponent?